src/top-bar/org-unit-select/approval-status-label.js

```diff
@@ -1,14 +1,36 @@
 import PropTypes from 'prop-types'
-// import React from 'react'
+import React from 'react'
+import { useSelectionContext } from '../../selection-context/index.js'
+import { getApprovalStateDisplayData } from '../../shared/approval-state/index.js'
+import classes from './approval-status-label.module.css'
+import { useApprovalStatuses } from './approval-statuses.js'
 
-const ApprovalStatusLabel = ({ id, label }) => {
-    // console.log(id, label)
+const ApprovalStatusLabel = ({ label, orgUnitId }) => {
+    const { workflow, period } = useSelectionContext()
+    const { getApprovalStatus } = useApprovalStatuses()
+    const approvalStatus = getApprovalStatus({
+        workflowId: workflow.id,
+        periodId: period.id,
+        orgUnitId,
+    })
 
-    return `ICON ${label}`
+    if (approvalStatus) {
+        const { displayName, icon: Icon } =
+            getApprovalStateDisplayData(approvalStatus)
+        return (
+            <div className={classes.container}>
+                <span title={displayName} className={classes.iconContainer}>
+                    <Icon />
+                </span>
+                {label}
+            </div>
+        )
+    }
```

Comment:
Does this take into account a loading state?

Reply:
This component has no loading state and does not set the correct `fill` style for icons based on the approval status as this PR was mostly for experimenting with how to store approval state.